dts/bindings/spi/spi-device.yaml

```diff
@@ -0,0 +1,37 @@
+#
+# Copyright (c) 2018, I-SENSE group of ICCS
+#
+# SPDX-License-Identifier: Apache-2.0
+#
+---
+title: SPI Device Base Structure
+id: spi-device
+version: 0.1
+
+description: >
+    This binding gives the base structures for all spi devices
+
+parent:
+    bus: spi
+
+properties:
+    compatible:
+      type: string
+      category: required
+      description: compatible strings
+    reg:
+      type: array
+      description: Chip select address of device
+      generation: define
+      category: required
```

Comment:
This should be "optional", no? Not all SPI devices support CS pins

Reply:
I haven't come across a SPI device without CS pin. I would suggest to keep it as required since SPI is a bus. In case of a device without CS pin, the SPI bus would have one device on "address" reg = <0>.

Reply:
Lpd8806 is one:

https://github.com/zephyrproject-rtos/zephyr/blob/master/drivers/led_strip/lpd880x.c#L127

Cheap, low pin count devices sometimes don't have a CS pin.